### PR TITLE
rfc: add rfc for framework migration

### PR DESCRIPTION
**Pull request by FeelyChau**
RFC for framework migration.

### CODE REVIEW

File: docs/rfc/0001-framework.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+- Start Date: 2020-11-25
```
Comment (yorkie) [inline docs/rfc/0001-framework.md:1]: 0001-framework-migration?

File: docs/rfc/0001-framework.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+- Target Major Version: 2.0
+- Reference Issues: (leave this empty)
+- Implementation PR: (leave this empty)
+- Author: @FeelyChau
+
+# Summary
+
+We have accumulated a lot of debt in the iteration process in Pipcook 1.0, and hope to solve these problems in 2.0.
```
Comment (yorkie) [inline docs/rfc/0001-framework.md:9]:
```suggestion
We have accumulated a lot of debt in the iteration process in Pipcook 1.x, and hope to solve these problems in 2.0.
```

File: docs/rfc/0001-framework.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    shuffle(array);
+    expect(array).not.toEqual([ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+    expect(array.sort()).toEqual([ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+  });
+}
+```
+
+## Integration test
+
+The integration test of pipbook 1.0 depends on the workflow configuration file. In a strict sense, this is not a rigorous integration testing method and hard to maintain. In addition, we do not assert any of the running process except exit code. This situation will be improved in pipbook 2.0 as follows:
```
Comment (yorkie) [inline docs/rfc/0001-framework.md:82]:
```suggestion
The integration test of 1.0 depends on the workflow configuration file. In a strict sense, this is not a rigorous integration testing method and hard to maintain. In addition, we do not assert any of the running processes except exit code. This situation will be improved in 2.0 as follows:
```

File: docs/rfc/0001-framework.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+* define integration test cases
+* use or develop an intergration test framework suitable for pipbook
```
Comment (yorkie) [inline docs/rfc/0001-framework.md:85]:
```suggestion
* use or develop an integration test framework suitable for Pipcook
```

File: docs/rfc/0001-framework.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+* extend the run environment
+
+# Framework migration for Daemon
+
+The framework of daemon will be moved from the original framework to `loopback 4` for the following reasons:
+
+* automatic code generation capability
+* framework maturity: the version of typescript cannot be configured in the original framework, resulting in the overall size of the pipbook package unable to be trimmed, and the test framework is constrained and cannot be switched
+* internationalization
```
Comment (yorkie) [inline docs/rfc/0001-framework.md:94]:
```suggestion
* i18n
```

**Review by yorkie — CHANGES_REQUESTED** [review]

**Comment by FeelyChau** [comment]
Updated, may I ask you to review again? @yorkie

**Review by yorkie — APPROVED** [review]

**Comment by rickycao-qy** [comment]
Is it better to make it more clear for pipboard migration. It seems only mentions parcel. Is it better to also replace ice.js and etc.

**Comment by SirM2z** [comment]
> The pipboard UI library will be migrated from the original framework to the [new framework](https://ant.design/), and the packaging tools will be migrated from `webpack` to `parcel`. The advantages of internationalization, familiarity of community contributors and convenience of use of `parcel` are mainly considered.

How about this ?

The pipboard UI library will be migrated from the ice.js to the [ant.design](https://ant.design/) for the following reasons:
- The icejs has no English documentation, so it is not easy to contribute
- Compared with the familiarity of community contributors, the ant.design library is higher than icejs
- Reduce the complexity of the framework to make it easy to maintain. Ice.js contains some complex functions that are unnecessary for pipboard, such as MPA (Multi-page application), SSR (Server-side rendering), Permission routing, etc.

And the packaging tools will be migrated from `webpack` to `parcel`. Mainly compared the configuration complexity, parcel is lower than webpack, easy to use.

**Comment by rickycao-qy** [comment]
> > The pipboard UI library will be migrated from the original framework to the [new framework](https://ant.design/), and the packaging tools will be migrated from `webpack` to `parcel`. The advantages of internationalization, familiarity of community contributors and convenience of use of `parcel` are mainly considered.
> 
> How about this ?
> 
> The pipboard UI library will be migrated from the ice.js to the [ant.design](https://ant.design/) for the following reasons:
> 
> * The icejs has no English documentation, so it is not easy to contribute
> * Compared with the familiarity of community contributors, the ant.design library is higher than icejs
> * Reduce the complexity of the framework to make it easy to maintain. Ice.js contains some complex functions that are unnecessary for pipboard, such as MPA (Multi-page application), SSR (Server-side rendering), Permission routing, etc.
> 
> And the packaging tools will be migrated from `webpack` to `parcel`. Mainly compared the configuration complexity, parcel is lower than webpack, easy to use.

This looks good to me

**Review by rickycao-qy — APPROVED** [review]

**Comment by yorkie** [comment]
@SirM2z May I ask you to update it to the current branch? That would be great for review.

**Comment by SirM2z** [comment]
> @SirM2z May I ask you to update it to the current branch? That would be great for review.

Already updated

File: docs/rfc/0001-framework-migration.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+- Start Date: 2020-11-25
+- Target Major Version: 2.0
+- Reference Issues: (leave this empty)
+- Implementation PR: (leave this empty)
+- Author: @FeelyChau
```
Comment (yorkie) [inline docs/rfc/0001-framework-migration.md:5]: @SirM2z Add you GitHub ID, too :)

File: docs/rfc/0001-framework-migration.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+The framework of daemon will be moved from the original framework to `loopback 4` for the following reasons:
+
+* automatic code generation capability
+* framework maturity: the version of typescript cannot be configured in the original framework, resulting in the overall size of the pipbook package unable to be trimmed, and the test framework is constrained and cannot be switched
+* i18n
+
+# Framework migration for Pipboard
+
+The pipboard UI library will be migrated from the ice.js to the [ant.design](https://ant.design/) for the following reasons:
```
Comment (yorkie) [inline docs/rfc/0001-framework-migration.md:98]: icejs or ice.js? We need to unify the framework name.

File: docs/rfc/0001-framework-migration.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+- The icejs has no English documentation, so it is not easy to contribute
```
Comment (yorkie) [inline docs/rfc/0001-framework-migration.md:99]: We need to detail the contributor class, in my mind, it's foreigners.

File: docs/rfc/0001-framework-migration.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+- Compared with the familiarity of community contributors, the ant.design library is higher than icejs
```
Comment (yorkie) [inline docs/rfc/0001-framework-migration.md:100]: ant.design is better to be linked as well :)

Comment (yorkie) [reply docs/rfc/0001-framework-migration.md:100]: higher is improper here, how about "the ant.design has more developers"?

File: docs/rfc/0001-framework-migration.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+- Reduce the complexity of the framework to make it easy to maintain. Ice.js contains some complex functions that are unnecessary for pipboard, such as MPA (Multi-page application), SSR (Server-side rendering), Permission routing, etc.
```
Comment (yorkie) [inline docs/rfc/0001-framework-migration.md:101]: Ditto, Ice.js, ice.js and icejs, we need to choose one.

**Comment by yorkie** [comment]
@SirM2z My review on your changes is done, good jobs!

File: docs/rfc/0001-framework-migration.md (hunk 10 above)
Comment (SirM2z) [reply docs/rfc/0001-framework-migration.md:101]: :) Use icejs, its [documentation](https://ice.work/docs/guide/migrate) is written like this

**Comment by SirM2z** [comment]
Already updated

**Review by yorkie — DISMISSED** [review]
LGTM

**Review by rickycao-qy — DISMISSED** [review]
LGTM

File: docs/rfc/0001-framework-migration.md (hunk 9 above)
Comment (SirM2z) [reply docs/rfc/0001-framework-migration.md]: Or use "ant.design is more widely used" ? The meaning of developers refers more to maintainers

Comment (yorkie) [reply docs/rfc/0001-framework-migration.md]: Yea, that seems good to me :)

**Review by yorkie — APPROVED** [review]

**Review by SirM2z — APPROVED** [review]

**Review by WenheLI — APPROVED** [review]
LGTM

**Review by rickycao-qy — APPROVED** [review]
LGTM

**Pull request merged**